dali/pipeline/operator/operator.cc

```diff
@@ -18,9 +18,9 @@ namespace dali {
 
 template <typename Backend>
 void OperatorBase::EnforceConstantBatchSize(const workspace_t<Backend> &ws) const {
-  for (int i = 0; i < ws.NumInput(); i++) {
-    DALI_ENFORCE(ws.GetInputBatchSize(i) == max_batch_size_,
-                 "Batch size has to be constant throughout the processing");
+  for (int i = 1; i < ws.NumInput(); i++) {
```

Comment:
I believe it's not possible. In case of the output, there are operators, which arbitrarily choose what batch size to produce, e.g. `CoinFlip` and `NormalDist`. The effect is that the output batch size does not need to match input - user provides no input, hence we assume that the operator should request for a batch size, but the operator actually produces batch size from the info in the ArgumentInput. The problem is, that there are so many unknowns...

Do you happen to know, what kind of cases we should guard against? We could write tests for them to know, if this shall work as it is or not.